nengo/spa/vocab.py

```diff
@@ -334,18 +347,32 @@ def transform_to(self, other, keys=None):
             terms in this list that do not exist in the Vocabularies will
             be created.
         """
-        if keys is None:
-            keys = list(self.keys)
-            for k in other.keys:
-                if k not in keys:
-                    keys.append(k)
-
-        t = np.zeros((other.dimensions, self.dimensions), dtype=float)
-        for k in keys:
-            a = self[k].v
-            b = other[k].v
-            t += np.outer(b, a)
-        return t
+        # If the parent vocabs of self and other are the same, then no
+        # transform is needed between the two vocabularies, so return an
+        # identity matrix.
+        my_parent = self if self.parent is None else self.parent
+        other_parent = other if other.parent is None else other.parent
+
+        if my_parent is other_parent:
+            return np.eye(self.dimensions)
+        else:
+            if keys is None:
+                if self.read_only and other.read_only:
+                    keys = [k for k in self.keys if k in other.keys]
+                elif self.read_only:
+                    keys = list(self.keys)
+                elif other.read_only:
+                    keys = list(other.keys)
+                else:
+                    keys = list(self.keys)
+                    keys.extend(other.keys)
```

Comment:
If a key shows up in both lists, then it'll now be added *twice*!  This should probably be done as a ```set```.